x-pack/plugins/security_solution/public/rules/routes.tsx

```diff
@@ -30,6 +31,11 @@ const RulesSubRoutes = [
     main: RuleDetailsPage,
     exact: true,
   },
+  {
+    path: '/rules/id/:detailName/:tabName',
```

Comment:
Let's specify all tabs that we support:

```
/rules/id/:detailName/:tabName(${RuleDetailTabs.alerts}|${RuleDetailTabs.exceptions}|${RuleDetailTabs.executionResults}|${RuleDetailTabs.executionEvents})
```

That would help us properly render 404 if users navigate to unexisting pages, like `/rules/id/rule_id/gobbledygook`.

Reply:
Updating code and cypress test for it.